nixos/modules/services/web-apps/wanderer.nix

```diff
@@ -0,0 +1,300 @@
+        '';
+      };
+
+      secretsFile = mkOption {
+        type = types.path;
+        example = "/run/secrets/wanderer";
+        description = ''
+          Secrets like {env}`MEILI_MASTER_KEY` and {env}`POCKETBASE_ENCRYPTION_KEY`
+          must be passed to the service without adding them to the world-readable Nix store.
+
+          This file needs to be available on the host on which `wanderer` is running.
+          {env}`POCKETBASE_ENCRYPTION_KEYE` must be exactly 32 characters long.
+          One way to generate such a secret is to use `openssl rand -hex 16`.
+
+          At a **minimum**, the contents of the file must look like this:
+          ```
+          MEILI_MASTER_KEY=...master key used by the wanderer backend and frontend, and by meilisearch...
+          POCKETBASE_ENCRYPTION_KEY=...encryption key used by the wanderer backend, exactly 32 chars long...
+          ```
+        '';
+      };
+
+      package = mkPackageOption pkgs "wanderer" { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.wanderer = {
+      description = "wanderer pocketbase db";
+      after = [
+        "network.target"
+        "meilisearch.service"
+      ];
+      wantedBy = [ "wanderer-web.service" ];
+
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+
+        StateDirectory = baseNameOf stateDir;
+        WorkingDirectory = stateDir;
+        Environment = [
+          "ORIGIN=${cfg.origin}"
+          "MEILI_URL=${cfg.services.meilisearch.url}"
+
+          "POCKETBASE_CRON_SYNC_SCHEDULE=\"${cfg.settings.syncSchedule}\""
+        ]
+        ++ attrValues (mapAttrs (name: value: name + "=" + builtins.toString value) cfg.extraConfig);
+        EnvironmentFile = cfg.secretsFile;
+        ExecStartPre =
+          pkgs.writeShellScript "wanderer-start-pre" # bash
+            ''
+              mkdir -p ${stateDir}/migrations
+              ln -sfn ${cfg.package}/share/initial_data ${stateDir}/migrations/initial_data
+            '';
+        ExecStart =
+          let
+            url = cfg.services.pocketbase.url;
+            scheme =
+              if lib.hasPrefix "https://" url then
+                "https"
+              else if lib.hasPrefix "http://" url then
+                "http"
+              else
+                throw "pocketbase url must start with http:// or https://";
+            host = lib.removePrefix "${scheme}://" url;
+          in
+          "${getExe cfg.package} serve --dir=\"${stateDir}\" --${scheme}=\"${host}\"";
+        Restart = "on-failure";
+      };
+    };
+
+    systemd.services.wanderer-web = {
+      description = "wanderer web app";
+      after = [
+        "wanderer.service"
+        "meilisearch.service"
+      ];
+      wantedBy = [ "multi-user.target" ];
+
+      path = [ pkgs.unionfs-fuse ];
```

Comment:
Assuming the package itself is using this program, is there a reason why it shouldn't injected in the package instead?

Reply:
Ah damnit, that's a leftover from an, uh, experiment. Removed.